javaslang/src/main/java/javaslang/control/Option.java

```diff
@@ -171,6 +171,19 @@
     boolean isEmpty();
 
     /**
+     * Runs a Java Runnable passed as parameter if this {@code Option} is empty.
+     *
+     * @param action a given Runnable to be run
+     * @return this {@code Option}
+     */
+    default Option<T> onEmpty(Runnable action) {
```

Comment:
Please add a null-check, we do it by convention for better readability of stack traces:

```java
Objects.requireNonNull(action, "action is null");
```